openfpga/src/fpga_bitstream/build_grid_bitstream.cpp

```diff
@@ -119,7 +120,14 @@ static void build_primitive_bitstream(
   }
 
   std::vector<bool> mode_select_bitstream;
-  if (true == physical_pb.valid_pb_id(primitive_pb_id)) {
+  if (pcf_mode_specified) {
```

Comment:
- This should not have a higher priority than a mapped block. It should have a higher priority than the default bitstream.

- If there are multiple mode_bits are specified for the pb_type with different offset, you should apply all of them.

- Also you should refer to LINE138 about how the offset should be applied.